namespaces/referencing-namespace.md

```diff
@@ -0,0 +1,194 @@
+# Referencing Namespace
+
+This namespace defines elements and attributes that may be used to provide the ability to reference and classify elements and namespaces.
+
+This document includes the elements as defined in the Refract specification.
+
+## Referencing and Identifying Elements
+
+These elements provide the means to identify elements with unique or reused qualifiers.
+
+### Ref (Element)
+
+The `ref` element provides a way to reference elements and namespaces with a URI. The context the `ref` element is found in determines what exactly it references. Additionally, the referenced
+
+#### Properties
+
+- `element`: ref (string, required)
+- `content` (string, required) - The URI of the referenced item
+
+### ID (Element)
+
+The `id` element provides a document-wide unique identifier. It MUST be unique for the entire document in which is found. When used as an attribute for another element, it defines the unique ID for that element, and MAY be used in the fragment identifier to link to that element.
+
+#### Properties
+
+- `element`: id (string, required)
+- `content` (string, required) - The unique identifier for an element
+
+#### Example
+
+This example shows the `id` element used as an attribute to provide a unique identifier.
+
+```json
+{
+  "element": "foo",
+  "attributes": {
+    "id": "my-foo"
+  },
+  "content": "bar"
+}
+```
+
+In this example, the `foo` element has an `id` of `my-foo`, which can be used to reference this specific instance of the `foo` element.
+
+### Roles (Element)
+
+The `roles` element allows for giving specific roles to element based on the context in which they are found. The roles MAY be reused throughout the document to provide more specific classification, uses, and roles of elements. It provides element reuse without the need for extending or creating new elements.
+
+#### Properties
+
+- `element`: roles (string, required)
+- `content` (array[string])
+
+#### Example
+
+This example shows the `roles` element being used as an attribute.
+
+```json
+{
+  "element": "user",
+  "attributes": {
+    "roles": ["customer"]
+  },
+  "content": [
+    {
+      "element": "id",
+      "content": 4
+    }
+  ]
+}
+```
+
+Using `roles`, the `user` element is given a more specific classification as a customer.
+
+## Namespacing
+
+Namespacing provides a way to do the following in a document:
+
+- Include elements directly from other documents
+- Define a prefix for referencing elements in defined namespaces
+
+This functionality exists to provide extendability and prevent name collisions in the event that two or more namespace specify the same element name.
+
+### Require (Element)
```

Comment:
Ref was simply a way to reference a particular element. Require works like require in many language environments in including elements into the namespace. Subtle difference, but probably good to remove `ref` for now since (as you can see) I didn't define it very clearly.